src/averageAges.js

```diff
@@ -54,8 +68,53 @@ function calculateWomenAverageAge(people, withChildren) {
  *
  * @return {number}
  */
-function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+function calculateAverageAgeDiff(people, onlyWithSon = false) {
+  let children = people.filter(person => person.mother !== null);
+
+  if (onlyWithSon) {
+    children = filterPersonSex(children, 'm');
+  }
+
+  const womenOnly = filterPersonSex(people, 'f');
+  const mothers = filterMothersNames(womenOnly, getMothersNames(children));
+
+  children = children.filter(child => (
+    mothers.some(mother => mother.name === child.mother)
+  ));
+
+  const CALCULATED_AVERAGE_AGE_DIFFERENCE = sumAgeDiff(children, mothers);
+
+  return calculateAverageAge(CALCULATED_AVERAGE_AGE_DIFFERENCE, children);
+}
+
+function filterPersonSex(people, sex) {
+  return people.filter(person => person.sex === sex);
+}
+
+function sumAge(people) {
+  return people.reduce((sum, person) => (
+    sum + (person.died - person.born)
+  ), 0);
+}
+
+function calculateAverageAge(calculatedAge, people) {
+  return Math.round(calculatedAge / people.length * 100) / 100;
+}
+
+function getMothersNames(people) {
```

Comment:
I like how you broke some functionality into different functions, but I affraid it would be better to simply reuse some code, especially with `sumAgeDiff` because it used only once